### ISSUE TITLE
Allow the exclusion of an author's username when browsing translation changes

### ISSUE THREAD

**Issue by ib** [WeblateOrg/weblate#12531]
### Describe the problem

It is very frustrating to browse translation changes when many changes have been made by user weblate. It is very hard to find changes made by real users among them.

### Describe the solution you would like

It would be useful to be able to exclude a username, especially the user weblate, when browsing all translation changes, either by specifying the user as !weblate or by a checkbox saying "exclude changes made by weblate" or something like that.

**Comment by 3ina** [comment]
Can you help me where to start? @nijel

**Comment by nijel** [comment]
The form for filtering changes is here:

https://github.com/WeblateOrg/weblate/blob/402c806df3ae12348085d1bcc67c4e701d33a47d/weblate/trans/forms.py#L2756-L2808

Filtering is here:

https://github.com/WeblateOrg/weblate/blob/402c806df3ae12348085d1bcc67c4e701d33a47d/weblate/trans/views/changes.py#L181-L188

I think the best approach would be to have an additional field "Exclude user" to filter out changes from user we would rather not see in the changes list.

**Comment by Siddharthgup** [comment]
Hi @ib & @nijel , Can I also take up this issue ?

**Comment by nijel** [comment]
All contributions are welcome! We don't assign issues to outside collaborators.

**Comment by wokepadia** [comment]
I looked into this issue, for all non-user changes, we have user_id set to null in our `trans_change` table. So all those changes do not have any user to filter for. Writing `anonymous` (which is a valid user) in Author name on the same page yields 0 results.

We can do a few things to solve this:

1. Backfill user_id with the weblate/anonymous user's user_id wherever it is null. 

This would be ideal since it feels odd to have a nullable user_id field. I am not sure whether it would correct to say that a change made by no user is same as saying that the change was made by the anonymous user. 

One thing to consider here is that the backfill migration might be long depending on the number of changes we expect to have in existing installations.

2. Use a special string to translate to the filter of user_id == null.

We can have a special string that the user can input on the frontend, e.g. `none`, and if that string is inserted in the author username or the exclude username field, we would include or exclude changes with `user_id=null` accordingly. The change will be made near https://github.com/wokepadia/weblate/blob/44cbd0bc0fb0864d0588167a5579a288db3d78d3/weblate/trans/forms.py#L2861 and https://github.com/wokepadia/weblate/blob/44cbd0bc0fb0864d0588167a5579a288db3d78d3/weblate/trans/views/changes.py#L34. We might need to adjust `clean_user` to not throw an error on that string and pass that string unchanged to ChangesView which then will insert the filter to the result.

Option 2 will be easier to implement, without worrying about migration times, but the solution there feels a bit hacky.

**Comment by nijel** [comment]
These are different - anonymous user is unauthenticated user from web, while `NULL` value indicates change done directly by Weblate. Most of these are now credited to a bot accounts, but maybe there are some actions left. I think there is an issue for this, but I can't find it right now.

**Comment by nijel** [comment]
https://github.com/WeblateOrg/weblate/pull/10097 handled this for add-ons, but there are definitely some events left (at least the ones triggered by webhooks).

**Comment by wokepadia** [comment]
In that case, can I assume the correct course of action for this issue is that first we will try to make sure that every action does by weblate has a user attached to it (without backfilling the data). Once that is done, we will come revisit this issue.

Or do we want an `Exclude changes made by Weblate` checkbox as an intermediate solution?

**Comment by ib** [comment]
It still doesn't seem to be possible to exclude ![Image](https://github.com/user-attachments/assets/f5536c02-edd8-4dfd-a138-ea6e148f3f2d), or am I missing something?

**Comment by nijel** [comment]
Yes, the plan is to get rid of that. On which events do you see that?

**Comment by ib** [comment]
> On which events do you see that?

When using Weblate's REST API to upload translations.

**Comment by nijel** [comment]
That should use the user who did that, can you please open a separate issue for that describing when this happens?

**Comment by ib** [comment]
> can you please open a separate issue for that

Done: #15298.
